website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
It seems to me that if we can determine whether a machine is an EC2 instance, the right thing to do here would be to merge this with the EC2 backend into simply an `aws` backend.

That way we can support two types of login (instance identity doc or signed request). If the instance identity doc is used then we return a nonce and use the existing whitelisting capabilities and if this method is used then no nonce is needed (but may still be used/desired to allow only a single entity on an EC2 box to auth). Either way, if the identity proves to be an EC2 instance, we can apply the full set of binds.

This would allow significant code reuse and allow this method to automatically gain support for further restrictions on EC2 instances.

Re the MFA bit, sure -- that would be great.